ui/src/components/template/TemplateList.js

```diff
@@ -46,25 +46,88 @@ const TableHeadStyled = styled.th`
     word-break: break-all;
 `;
 
+const TitleDiv = styled.div`
+    font: 600 15px HelveticaNeue-Reg, Helvetica, Arial, sans-serif;
+    margin-bottom: 20px;
+    margin-top: 40px;
+`;
+
 export default class TemplateList extends React.Component {
     constructor(props) {
         super(props);
         this.api = props.api;
         this.onCancelUpdateTemplate = this.onCancelUpdateTemplate.bind(this);
         this.reloadTemplates = this.reloadTemplates.bind(this);
+        this.showError = this.showError.bind(this);
         this.closeModal = this.closeModal.bind(this);
+        let serverTemplateListCopy = [];
+        if (props.serverTemplateDetails) {
+            serverTemplateListCopy = props.serverTemplateDetails.map((x) => x);
+
+            if (props.domainTemplateDetails) {
+                let domainTemplatesArray = props.domainTemplateDetails.map(
+                    (template) => {
+                        return template.templateName;
+                    }
+                );
+
+                let domainTemplatesSet = new Set(domainTemplatesArray);
+                serverTemplateListCopy = serverTemplateListCopy.filter(
+                    (template) => !domainTemplatesSet.has(template.templateName)
+                );
+            }
+        }
+
         this.state = {
             list: props.domainTemplateDetails || [],
+            serverTemplateDetails: serverTemplateListCopy || [],
             successMessage: '',
+            errorMessage: '',
+            showError: false,
+            showSuccess: false,
         };
     }
 
+    showError(errorMessage) {
+        this.setState({
+            showError: true,
+            errorMessage: errorMessage,
+        });
+        // this is to close the failure alert
+        setTimeout(
+            () =>
+                this.setState({
+                    showError: false,
+                    errorMessage: '',
+                }),
+            MODAL_TIME_OUT
```

Comment:
we shouldn't use timed close for errors. It should persist on the screen until user actively dismisses it.

Reply:
Fixed